### PR TITLE
Cleanup and simplify the logics of fillna

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR propose to simplify fillna and the tests of fillna.

### CODE REVIEW

File: databricks/koalas/typedef.py [hunk 1]
```diff
@@ -170,18 +170,6 @@ def _check_compatible(arg, sig_arg: X):
     assert False, (arg, sig_arg)
 
 
-def dict_sanitizer(input_dict):
-    """
-    This function checks if elements inside a given dict are supported by Spark.
-    :param input_dict: dict
-    :raises TypeError if any element is not supported by Spark
-    """
-    _possible_type_values = [int, str, float, np.float64, np.float, bool]
```
Comment (HyukjinKwon) [inline databricks/koalas/typedef.py]: `np.float64` and `np.float` seems inheriting `float`:

```python
>>> import numpy as np
>>> np.float64(1.0)
1.0
>>> type(np.float64(1.0))
<class 'numpy.float64'>
>>> isinstance(np.float64(1.0), float)
True
>>> type(np.float(1.0))
<class 'float'>
>>> isinstance(np.float(1.0), float)
True
```

That seems why it works with Py4J.

Comment (thunterdb) [reply databricks/koalas/typedef.py]: Ah, this must be a recent change, I recall having some issues with this before.

Comment (thunterdb) [reply databricks/koalas/typedef.py]: what does it give with `np.float32` and `np.float16`?

Comment (HyukjinKwon) [reply databricks/koalas/typedef.py]: Interesting :D

```python
>>> import numpy as np
>>> type(np.float32(1.0))
<class 'numpy.float32'>
>>> isinstance(np.float32(1.0), float)
False
>>> type(np.float16(1.0))
<class 'numpy.float16'>
>>> isinstance(np.float16(1.0), float)
False
```

**Comment by HyukjinKwon** [comment]
This is just a clean up. I merged.

**Comment by garawalid** [comment]
Nice clean up :+1:

**Pull request merged**